apps/admin-gui/src/app/admin/pages/admin-page/admin-audit-log/admin-audit-log.component.ts

```diff
@@ -1,16 +1,49 @@
-import { Component } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
 import { TABLE_AUDIT_MESSAGES } from '@perun-web-apps/config/table-config';
+import { SelectionModel } from '@angular/cdk/collections';
+import { AuditMessage, AuditMessagesManagerService } from '@perun-web-apps/perun/openapi';
 
 @Component({
   selector: 'app-admin-audit-log',
   templateUrl: './admin-audit-log.component.html',
   styleUrls: ['./admin-audit-log.component.scss'],
 })
-export class AdminAuditLogComponent {
+export class AdminAuditLogComponent implements OnInit {
   tableId = TABLE_AUDIT_MESSAGES;
   refresh = false;
 
+  selection = new SelectionModel<AuditMessage>(true, []);
+
+  selectedEvents: string[] = [];
+  eventOptions: string[] = [];
+
+  constructor(private auditMessagesManagerService: AuditMessagesManagerService) {}
+
+  ngOnInit(): void {
+    this.auditMessagesManagerService.findAllPossibleEvents().subscribe((res) => {
+      this.eventOptions = res.sort();
+    });
+  }
+
   refreshTable(): void {
     this.refresh = !this.refresh;
   }
+
+  searchFunction = (entity: string): string => entity;
+
+  mainTextFunction = (entity: string): string => entity;
+
+  secondaryTextFunction = (_: string): string => '';
```

Comment:
As I mentioned, define these in the search select specification. There are also examples of methods that don't need the parameter.